contracts/asset-proxy/contracts/src/bridges/DexForwarderBridge.sol

```diff
@@ -0,0 +1,210 @@
+/*
+
+  Copyright 2020 ZeroEx Intl.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+*/
+
+pragma solidity ^0.5.9;
+pragma experimental ABIEncoderV2;
+
+import "@0x/contracts-erc20/contracts/src/interfaces/IERC20Token.sol";
+import "@0x/contracts-erc20/contracts/src/LibERC20Token.sol";
+import "@0x/contracts-exchange-libs/contracts/src/IWallet.sol";
+import "@0x/contracts-exchange-libs/contracts/src/LibMath.sol";
+import "@0x/contracts-utils/contracts/src/LibBytes.sol";
+import "@0x/contracts-utils/contracts/src/LibSafeMath.sol";
+import "../interfaces/IERC20Bridge.sol";
+
+
+// solhint-disable space-after-comma, indent
+contract DexForwarderBridge is
+    IERC20Bridge,
+    IWallet
+{
+    using LibSafeMath for uint256;
+
+    /// @dev Data needed to reconstruct a bridge call.
+    struct BridgeCall {
+        address target;
+        uint256 inputTokenAmount;
+        uint256 outputTokenAmount;
+        bytes bridgeData;
+    }
+
+    /// @dev Intermediate state variables used by `bridgeTransferFrom()`, in
+    ///      struct form to get around stack limits.
+    struct TransferFromState {
+        address inputToken;
+        uint256 initialInputTokenBalance;
+        uint256 callInputTokenAmount;
+        uint256 callOutputTokenAmount;
+        uint256 totalInputTokenSold;
+        BridgeCall[] calls;
+    }
+
+    event DexForwarderBridgeCallFailed(
+        address indexed target,
+        address inputToken,
+        address outputToken,
+        uint256 inputTokenAmount,
+        uint256 outputTokenAmount
+    );
+
+    /// @dev Executes a series of calls, forwarding .
+    /// @param outputToken The token being bought.
+    /// @param to The recipient of the bought tokens.
+    /// @param bridgeData The abi-encoeded input token address.
+    /// @return success The magic bytes if successful.
+    function bridgeTransferFrom(
+        address outputToken,
+        address /* from */,
+        address to,
+        uint256 /* amount */,
+        bytes calldata bridgeData
+    )
+        external
+        returns (bytes4 success)
+    {
+        TransferFromState memory state;
+        (
+            state.inputToken,
+            state.calls
+        ) = abi.decode(bridgeData, (address, BridgeCall[]));
+
+        state.initialInputTokenBalance = IERC20Token(state.inputToken).balanceOf(address(this));
```

Comment:
might as well use LibERC20Token for balances

Reply:
`LibERC20Token.balanceOf()` swallows reverts. Not my jam.